patchwork-events-entity/src/main/java/net/patchworkmc/mixin/event/entity/MixinCraftingResultSlot.java

```diff
@@ -0,0 +1,58 @@
+/*
+ * Minecraft Forge, Patchwork Project
+ * Copyright (c) 2016-2020, 2019-2020
+ *
+ * This library is free software; you can redistribute it and/or
+ * modify it under the terms of the GNU Lesser General Public
+ * License as published by the Free Software Foundation version 2.1
+ * of the License.
+ *
+ * This library is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU
+ * Lesser General Public License for more details.
+ *
+ * You should have received a copy of the GNU Lesser General Public
+ * License along with this library; if not, write to the Free Software
+ * Foundation, Inc., 51 Franklin Street, Fifth Floor, Boston, MA  02110-1301  USA
+ */
+
+package net.patchworkmc.mixin.event.entity;
+
+import org.spongepowered.asm.mixin.Final;
+import org.spongepowered.asm.mixin.Mixin;
+import org.spongepowered.asm.mixin.Shadow;
+import org.spongepowered.asm.mixin.injection.At;
+import org.spongepowered.asm.mixin.injection.Inject;
+import org.spongepowered.asm.mixin.injection.callback.CallbackInfo;
+import org.spongepowered.asm.mixin.injection.At.Shift;
+
+import net.minecraft.container.CraftingResultSlot;
+import net.minecraft.entity.player.PlayerEntity;
+import net.minecraft.inventory.CraftingInventory;
+import net.minecraft.item.ItemStack;
+
+import net.patchworkmc.impl.event.entity.PlayerEvents;
+
+@Mixin(CraftingResultSlot.class)
+public abstract class MixinCraftingResultSlot {
+	@Shadow
+	@Final
+	private PlayerEntity player;
+
+	@Shadow
+	@Final
+	private CraftingInventory craftingInv;
+
+	@Inject(method = "onCrafted(Lnet/minecraft/item/ItemStack;)V",
+			at = @At(
+					value = "INVOKE",
+					shift = Shift.AFTER,
+					ordinal = 0,
+					target = "net/minecraft/item/ItemStack.onCraft(Lnet/minecraft/world/World;Lnet/minecraft/entity/player/PlayerEntity;I)V"
+					)
```

Comment:
This indentation seems a bit odd, but I'm guessing it's correct based on checkstyle passing...

Reply:
I dont want to squash tooo much thing on a single line, this may be easier to read and understand

Reply:
i think what leo is getting at is that the indentation is a bit inconsistent, specifically the closing brace that was selected for this comment. i.e. I'd probably write it as either:
```java
    @Inject(
        method = "onCrafted(Lnet/minecraft/item/ItemStack;)V",
		at = @At(
			value = "INVOKE",
			shift = Shift.AFTER,
			ordinal = 0,
			target = "net/minecraft/item/ItemStack.onCraft(Lnet/minecraft/world/World;Lnet/minecraft/entity/player/PlayerEntity;I)V"
		)
    )
```

or:
```java
    @Inject(method = "onCrafted(Lnet/minecraft/item/ItemStack;)V", at = @At(
		value = "INVOKE",
		shift = Shift.AFTER,
		ordinal = 0,
		target = "net/minecraft/item/ItemStack.onCraft(Lnet/minecraft/world/World;Lnet/minecraft/entity/player/PlayerEntity;I)V"
	))
```